lkmpg.tex

```diff
@@ -925,6 +925,13 @@ \subsection{chardev.c}
 Don't worry if you don't see what we do with the data we read into the buffer; we don't do much with it.
 We simply read in the data and print a message acknowledging that we received it.
 
+In the multiple-threaded environment, without any protection, concurrent access to the same memory may lead to the race condition, and will not preserve the performance.
+In the kernel module, this problem may happen due to multiple instances accessing the shared resources.
+Therefore, a solution is to enforce the exclusive access.
+We use atomic Compare-And-Swap (CAS), the single atomic operation, to checks out whether the file is currently open by someone.
```

Comment:
Change "to checks out" to
> to determine

You shall boost your English grammar skills by regular writing.